### ISSUE TITLE
Quadratic voting interface for delegations

### ISSUE THREAD

**Issue by santisiri** [DemocracyEarth/sovereign#247]
An early prototype for enabling quadratic voting features applied to delegations on Sovereign.

**Comment by virgile-dev** [comment]
Hey @santisiri 
Why only on delegations? I would have implemented it for voting as well. 
What do you think ?

**Comment by virgile-dev** [comment]
The few papers I read is often applied to voting arguing that it is the best way "to supply better valuations that aggregate private information of dispersed multitudes." ( [Eric Posner](http://ericposner.com/quadratic-voting/)

I'll get a crack at mocking up something for delegation, but I don't why voting on issue shoudn't benefit from it mitigates the impact of the powerful.

**Comment by virgile-dev** [comment]
This a quick one just to make sure I understood the principle.
![quadratic-delegation](https://user-images.githubusercontent.com/11473995/38145443-a64e29e8-3449-11e8-8cde-37eb91ee1c33.png)

**Comment by santisiri** [comment]
I like the mockup, yet I think we need to convey more clearly why the additional cost. Reminds me of how ETH uses Gas for computations. 

The idea  I had in mind is simply not allowing intermediate values in the slider.. the slider would go from 1, 2, 4, 8, 16 and have some kind of gradient to give the idea that it is a quadratic vote this time (not a standard one).

**Comment by virgile-dev** [comment]
Ok ! will work on another iteration based on those principles. 
I think maybe we should still have a little debate on weather it's a good idea to implement quadratic voting on ballots. What are you thoughts @DemocracyEarth/sovereign-development ?

**Comment by AlexJupiter** [comment]
@santisiri @virgile-dev I don't think it should be understated how important and difficult the implementation of quadratic voting will be. We can't shy away from the fact that this is going to add *another* degree of complexity to an already complex product. 

These complexities we will solve, however it's going to take time, and we have to pick our battles carefully at this stage.

A couple of questions I have before offering a full opinion:
- Has there been any pilots or any research into the implementation of quadratic voting? 
- If we were to work on this now, it would take resources away from the higher priority items we have identified in our [product roadmap](https://github.com/DemocracyEarth/sovereign/projects/11
). Where would this sit in terms of priorities?

**Comment by virgile-dev** [comment]
@AlexJupiter The only quadratic voting implementation I know of is this one (no demo available)
https://collectivedecisionengines.com/quadratic-voting.html

I don't think we should tackle this right now. Let's prioritize the `/peer` view

**Comment by AlexJupiter** [comment]
> I don't like thinking that I will be taxed for doing the responsible and effective choice of admitting my own complete ignorance on most matters in the world and delegating to the appropriate people that I choose and trust.

I think this is an excellent point in my opinion!

**Comment by santisiri** [comment]
> What is a monopoly in a liquid democracy? A leadership backed by sovereign individuals who can revoke their delegations at any time? If a populist leader amasses too much power and than doesn't deliver to his sovereigns' (let's resignify sovereign to the old "ordinary" citizen) satisfaction, than he will be removed from power in a minute. Look at Trump's approval ratings and imagine that reflected in delegations. I'm not worried. We should create a system that treats Sovereigns like Sovereigns and trust and empower them with freedom, flexibility, liquidity.

Check the experience of the [Pirate Party in Germany](http://www.spiegel.de/international/germany/liquid-democracy-web-platform-makes-professor-most-powerful-pirate-a-818683.html) that we refer in the paper. That's a monopoly in a liquid democracy. The purpose of democracy is to keep an incentive for people to participate. That monopoly led to the end of that experiment. The example of Trump's approval rating is interesting but I don't see exactly how it connects to the proposed idea. In the case of our type of democracy (which is token based), could mean having a Zuckerberg.. his approval ratings also zuck (pun intended) but he's still controlling the network. 

Also the purpose is enabling a conscious thought on how yparticipants state preferences in the system. Quadratic voting is able to capture high-signal regarding where you are putting your priorities as you participate in the collective decision process. Which can help to the idea of having delegates empowered for specific topics (and avoid great generalists running the show as is politics today). 

> I don't like thinking that I will be taxed for doing the responsible and effective choice of admitting my own complete ignorance on most matters in the world and delegating to the appropriate people that I choose and trust. I also am uncomfortable with the idea of taxing users unless its a necessary fee for the maintenance of the system.

That's why I'm thinking about this only on the realm of delegations.

> Looking forward to see the counter arguments, but as of now I personally think quadratic voting is a bad idea.

I put thumbs down on your post because I don't believe in bad ideas. Anyway, I think this is a toggle feature that has little cost to implement but valuable insight for research. Check the [paper](http://ericposner.com/quadratic-voting/) if you haven't already.

**Comment by santisiri** [comment]
> @santisiri @virgile-dev I don't think it should be understated how important and difficult the implementation of quadratic voting will be. I am concerned that this is going to add another degree of complexity to an already complex product.

I think this is a good concern. I haven't advanced on its development so we could push it forward to a future milestone as we'll focus on Venezuela.

**Comment by paulamlb** [comment]
>The example of Trump's approval rating is interesting but I don't see exactly how it connects to the proposed idea. In the case of our type of democracy (which is token based), could mean having a Zuckerberg.. his approval ratings also zuck (pun intended) but he's still controlling the network. 

What I am implying here is that in the way I see it, in a liquid democracy zuck's control of the network would be pending on his approval ratings (liquid delegations). No one gets a blank check to do whatever they want, delegates can have their sovereigns revoke their delegation at any time. So the ability to revoke can also be considered a mechanism against the formation of monopolies.

From [the article you linked](http://www.spiegel.de/international/germany/liquid-democracy-web-platform-makes-professor-most-powerful-pirate-a-818683.html):

<img width="684" alt="screen shot 2018-04-20 at 21 44 35" src="https://user-images.githubusercontent.com/18194034/39079219-245fd32c-44e4-11e8-81be-fe01926cb5f8.png">

The Pirate Party example also demonstrates how users were delegating in a conscious way and "promptly withdrew their votes" once Haase voted in a way that they weren't happy with. Haase doesn't strike me as the populist leader, his delegators seemed to be critical thinkers who were monitoring him closely, and I am actually encouraged to see that this system enables someone that is a profoundly engaged scholar such as him to arise to power, which usually doesn't happen in representative democracy. LD is about empowering the smartest minds in the system, and one might argue that it did just that in Germany. 

Now, even if that is the case, that is still a very specific universe and we can't affirm that this is how things will always play out. Nevertheless I think that before discussing quadratic voting we might want to investigate how liquid democracy performs in terms of formation of monopolies, to see if it is really an issue or not. Also think about how quadratic voting performs with LD. Student-led pilots might be greatly informative in that sense.

**Comment by santisiri** [comment]
I think the angle on...

- Revoke as a control mechanism
- Ability to not be conditioned if you really want to delegate power to someone.

... are strong arguments. 

The goal is to prevent formation of monopolies. Revoking allows it but it focuses on the punishment side of things if the empowerment goes wrong. Whereas quadratic voting directly prevents it. I think we can keep discussing this feature and put it on hold in terms of implementation. It would be interesting nonetheless to find a context where we can test it out.

**Comment by paulamlb** [comment]
I have introduced the topic to the students that I am working with. We could potentially deploy a separate version with quadratic voting for their pilots starting in the next school year and then compare the emerging dynamics in terms of formation of monopolies with the main version. 

I wonder if it would be possible to eventually merge the two, so we don't miss on the network effects of the work done by the students?

**Comment by marc-wagner** [comment]
Hi, a thought on quadratic voting in a decentralized, open network (i.e a blockchain): 
for quadratic voting to work, IMHO you need to enforce delegation through 
this mechanism. otherwise, people will coordinate outside of the DE framework
(through smart contracts similar to bitcoin tumblers or multisig)  
and cast individual votes, achieving a bigger impact than people playing
the quadratic delegation game. Please note I am not talking about a sybil attack
here. Any thoughts on how this could be enforced?

**Comment by marc-wagner** [comment]
Rather than trying to block such outside collusion, 
an easier way may well be to support it

 through an 'everyone' contract that impersonates anonymous masses at ballots, with a weight =  sum of individual votes on both sides.

**Comment by doctor-gonzo** [comment]
I am a bit confused about DEF's definition of Quadratic Voting, I opened an issue in the paper section and would appreciate any input 
https://github.com/DemocracyEarth/paper/issues/369

**Comment by santisiri** [comment]
hi @doctor-gonzo, yes, we are aware of that. our view is a simplification of the original QV model that deals with the accounting of votes and their price separately.. our aim was to simply make a synthesis of both.
